src/main/java/seedu/address/model/person/DeadlineList.java

```diff
@@ -46,6 +78,14 @@ public String listFormat() {
         return result.toString();
     }
 
+    /**
+     * Gives the size of this deadline list, considering only deadlines given by user.
+     */
+    public int size() {
+        return this.deadlines.size() == 1 && this.deadlines.get(0).equals(EMPTY_DEADLINE) ? 0
+                : this.deadlines.size();
+    }
```

Comment:
I think this logic seems a bit weird. I'm not sure if I'm misunderstanding this, but can't we have `EMPTY_DEADLINE` return 0 on `emptyDeadline#size`?

Reply:
What I understand of the implementation of Deadline and DeadlineList is that a DeadlineList with no Deadlines added by the user still contains the `EMPTY_DEADLINE` object, as you can see in the default constructor. However, my `size` method wants to check the number of Deadlines supplied by the user, which excludes `EMPTY_DEADLINE`, so I can't simply return the size of the internal ArrayList 😔 .

I'm not sure if I understand your suggestion right, but if it is to have an `EMPTY_DEADLINELIST` containing an `EMPTY_DEADLINE` as the static variable, and running a comparison check with it and returning 0 if true, then I think that will indeed clean up this method a bit. 🤔

Reply:
I understand where Ezekiel is coming from and I'm guessing this is what he meant
```suggestion
    public int size() {
        return this.deadlines.get(0).equals(EMPTY_DEADLINE) ? 0
                : this.deadlines.size();
    }
```

Reply:
I agree that this solution is better!